protocols/gossipsub/src/behaviour.rs

```diff
@@ -661,7 +661,8 @@ where
             && self.forward_msg(&msg_id, raw_message.clone(), None, HashSet::new())?;
 
         let mut recipient_peers = HashSet::new();
-        if let Some(set) = self.topic_peers.get(&topic_hash) {
+        let set: Vec<_> = self.peer_topics.keys().collect();
```

Comment:
This change now gets all peers regardless of which topic they are subscribed to

Reply:
Yes, true, I think this is wrong here even for what I was attempting to.

Basically what I see is that `self.topic_peers.get(&topic_hash)` is `None` when you try to publish a msg from a peer which is not aware of a peer subscribed to that topic. Thus I thought the `flood-publish` is what would enable such a msg in such case to still be sent to connected peers so it eventually finds its way through.